endgame_simulations/endgame_simulation.py

```diff
@@ -249,14 +255,21 @@ def iter_run(
         end_time: float,
         sampling_interval: float | None = None,
         sampling_years: list[float] | None = None,
+        inclusive: bool = False,
     ) -> Iterator[State]:
         while self.simulation.state.current_time < end_time:
             # Invariant: current params are applied at this point
             if self.next_params_index < len(self._param_set):
                 time, next_params = self._param_set[self.next_params_index]
-                next_stop = min(time, end_time)
+                if inclusive:
```

Comment:
If you defined something like `inclusive_adjustment = self.simulation._delta_time if inclusive else 0.0` then you don't need any extra branching beyond what was already there I think.

Reply:
```py
inclusive_adjustment = self.simulation._delta_time if inclusive else 0.0
if self.next_params_index < len(self._param_set):
     time, next_params = self._param_set[self.next_params_index]
     next_stop = min(time, end_time + inclusive_adjustment)
else:
     next_stop = end_time + inclusive_adjustment
     next_params = None